deeppavlov/skills/aiml_skill/aiml_skill.py

```diff
@@ -0,0 +1,139 @@
+    AIML uses directory with AIML scripts which are loaded at initialization and used as patterns
+    for answering at each step.
+    """
+
+    def __init__(self,
+                 path_to_aiml_scripts: Union[str, None] = None,
+                 positive_confidence: float = 0.66,
+                 null_response: str = "I don't know",
+                 null_confidence: float = 0.33,
+                 **kwargs
+                 ) -> None:
+        """
+        Construct skill:
+            read AIML scripts,
+            load AIML kernel
+
+        Args:
+            path_to_aiml_scripts: string path to folder with AIML scripts
+            null_response: Response string to answer if no AIML Patterns matched
+            positive_confidence: The confidence of response if response was found in AIML scripts
+            null_confidence: The confidence when AIML scripts has no rule for responding and system returns null_response
+        """
+
+        if not path_to_aiml_scripts:
+            cur_dir = Path(__file__).absolute().parent
+            self.path_to_aiml_scripts = cur_dir.joinpath("aiml_scripts")
+            log.warning(f"path_to_aiml_scripts is not provided, use default path: `{self.path_to_aiml_scripts}`")
+        else:
+            self.path_to_aiml_scripts = Path(path_to_aiml_scripts)
+            log.info(f"path_to_aiml_scripts is: `{self.path_to_aiml_scripts}`")
+
+        self.positive_confidence = positive_confidence
+        self.null_confidence = null_confidence
+        self.null_response = null_response
+        self.kernel = aiml.Kernel()
+        # to block AIML output:
+        self.kernel._verboseMode = False
+        self._load_scripts()
+
+    def _load_scripts(self) -> None:
+        """
+        Scripts are loaded recursively from files with extensions .xml and .aiml
+        Returns: None
+
+        """
+        # learn kernel to all aimls in directory tree:
+        all_files = sorted(self.path_to_aiml_scripts.rglob('*.*'))
+        for each_file_path in all_files:
+            if each_file_path.suffix in ['.aiml', '.xml']:
+                # learn the script file
+                self.kernel.learn(str(each_file_path))
+
+    def process_step(self, utterance_str: str, user_id: any) -> Tuple[str, float]:
+        response = self.kernel.respond(utterance_str, sessionID=user_id)
+        # here put your estimation of confidence:
+        if response:
+            # print(f"AIML responds: {response}")
+            confidence = self.positive_confidence
+        else:
+            # print("AIML responses silently...")
+            response = self.null_response
+            confidence = self.null_confidence
+        return response, confidence
+
+    def _generate_user_id(self) -> str:
+        """
+        Here you put user id generative logic if you want to implement it in the skill.
+
+        Although it is better to delegate user_id generation to Agent Layer
+        Returns: int
```

Comment:
Is it `str` or `int`?

Reply:
Really it is uuid.UUID object =)
Fixed docs